src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Apprenticeships.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataManagement.Schemas;
+using Dfc.CourseDirectory.Core.Models;
+
+namespace Dfc.CourseDirectory.Core.DataManagement
+{
+    public partial class FileUploadProcessor
+    {
+        public async Task<SaveFileResult> SaveApprenticeshipFile(Guid providerId, Stream stream, UserInfo uploadedBy)
+        {
+            CheckStreamIsProcessable(stream);
+
+            if (await FileIsEmpty(stream))
+            {
+                return SaveFileResult.EmptyFile();
+            }
+
+            if (!await LooksLikeCsv(stream))
+            {
+                return SaveFileResult.InvalidFile();
+            }
+
+            var (fileMatchesSchemaResult, missingHeaders) = await FileMatchesSchema<CsvApprenticeshipRow>(stream);
```

Comment:
I think this part is on a different story (if not it would be good to flesh out `CsvApprenticeshipRow` so it's complete)

Reply:
done